### ISSUE TITLE
Accuracy of XGBClassifier of disable mode

### ISSUE THREAD

**Issue by nayeong-cryptolab** [zama-ai/concrete-ml#819]
## Summary

The accuracy of XGBClassifier of disable mode is lower than the accuracy of simulate/execute mode.

## Description

- versions affected: 1.6.1
- python version: 3.8
- config (optional: HW, OS): Ubuntu 20.04.6 LTS

Step by step procedure someone should follow to trigger the bug:

<details><summary>minimal POC to trigger the bug</summary>
<p>

```python
from concrete.ml.sklearn import XGBClassifier

model = XGBClassifier(
        max_depth=2, n_estimators=2, random_state=0, n_bits=4
 )
model.fit(X_train, y_train)

y_pred_plain = model.predict(X_test, fhe="disable")

model_circuit = model.compile(X_train)
model_circuit.keygen(force=True)
y_pred_enc = model.predict(X_test, fhe="execute")
```

Accuracy tested with the scikit-learn iris dataset is as follows:
```
Plaintext ACC: 30.00%
Encrypted ACC: 93.33%
```

</p>
</details>

**Comment by andrei-stoian-zama** [comment]
hi,

I tried to reproduce this way:


```

from concrete.ml.sklearn import XGBClassifier
from sklearn.datasets import load_iris
from sklearn.model_selection import train_test_split
from sklearn.metrics import accuracy_score

model = XGBClassifier(
        max_depth=2, n_estimators=2, random_state=0, n_bits=4
 )

X, y = load_iris(return_X_y=True)
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2)

model.fit(X_train, y_train)

y_pred_plain = model.predict(X_test, fhe="disable")

model_circuit = model.compile(X_train)
model_circuit.keygen(force=True)
y_pred_enc = model.predict(X_test, fhe="execute")

print(accuracy_score(y_pred_plain, y_test))
print(accuracy_score(y_pred_enc, y_test))

```

but the accuracy is the same plain vs enc

**Comment by nayeong-cryptolab** [comment]
Hi, thank you for the quick response.

The accuracy is the same plain vs enc when I have tried in a different environment.
